test/reduce.js

```diff
@@ -116,6 +116,11 @@ module.exports = function reduce_test(testcase, minify_options, reduce_options)
                 ][ ((node.start._permute += step) * steps | 0) % 3 ];
             }
             else if (node instanceof U.AST_BlockStatement) {
+                if (in_list) {
+                    node.start._permute++;
+                    CHANGED = true;
+                    return List.splice(node.body);
+                }
                 if (node.body.length === 1) {
                     node.start._permute++;
                     CHANGED = true;
```

Comment:
`return node.body[0];  // each child is an AST_Statement`   (I wanted to annotate after this line)

Maybe you can get rid of this condition altogether since the `List.splice` permutation already takes care of it. What you wrote is what I wanted to do in the first place. Try removing it and see what happens. It's all trial and error.

Reply:
Done.

Reply:
I tried the latest version of this PR on my 10 failing `evaluate` xor `^` test cases and reduced code quality improved and superfluous blocks were elided. Timings only went up by 10% which is perfectly acceptable given the improved generated code.